sktime/forecasting/compose/_ensemble.py

```diff
@@ -135,38 +139,51 @@ def _fit(self, y, X=None, fh=None):
         _, forecasters = self._check_forecasters()
-        self.regressor_ = check_regressor(
-            regressor=self.regressor, random_state=self.random_state
-        )
 
-        # get training data for meta-model
-        if X is not None:
-            y_train, y_test, X_train, X_test = temporal_train_test_split(
-                y, X, test_size=self.test_size
+        if self.method is None:
+            self.regressor_ = check_regressor(
+                regressor=self.regressor, random_state=self.random_state
             )
-        else:
-            y_train, y_test = temporal_train_test_split(y, test_size=self.test_size)
-            X_train, X_test = None, None
-
-        # fit ensemble models
-        fh_regressor = ForecastingHorizon(y_test.index, is_relative=False)
-        self._fit_forecasters(forecasters, y_train, X_train, fh_regressor)
-        X_meta = pd.concat(self._predict_forecasters(fh_regressor, X_test), axis=1)
-
-        # fit meta-model (regressor) on predictions of ensemble models
-        # with y_test as endog/target
-        self.regressor_.fit(X=X_meta, y=y_test)
-
-        # check if regressor is a sklearn.Pipeline
-        if isinstance(self.regressor_, Pipeline):
-            # extract regressor from pipeline to access its attributes
-            self.weights_ = _get_weights(self.regressor_.steps[-1][1])
-        else:
-            self.weights_ = _get_weights(self.regressor_)
-
-        # fit forecasters with all data
-        self._fit_forecasters(forecasters, y, X, fh)
 
+            # get training data for meta-model
+            if X is not None:
+                y_train, y_test, X_train, X_test = temporal_train_test_split(
+                    y, X, test_size=self.test_size
+                )
+            else:
+                y_train, y_test = temporal_train_test_split(y, test_size=self.test_size)
+                X_train, X_test = None, None
+
+            # fit ensemble models
+            fh_regressor = ForecastingHorizon(y_test.index, is_relative=False)
+            self._fit_forecasters(forecasters, y_train, X_train, fh_regressor)
+            X_meta = pd.concat(self._predict_forecasters(fh_regressor, X_test), axis=1)
+
+            # fit meta-model (regressor) on predictions of ensemble models
+            # with y_test as endog/target
+            self.regressor_.fit(X=X_meta, y=y_test)
+
+            # check if regressor is a sklearn.Pipeline
+            if isinstance(self.regressor_, Pipeline):
+                # extract regressor from pipeline to access its attributes
+                self.weights_ = _get_weights(self.regressor_.steps[-1][1])
+            else:
+                self.weights_ = _get_weights(self.regressor_)
+            self._fit_forecasters(forecasters, y, X, fh)
+        elif self.method == "inv_var":
+            # get in-sample forecasts
+            self._fit_forecasters(forecasters, y, X, fh)
+            fh_insample = ForecastingHorizon(y.index, is_relative=False)
```

Comment:
I am not sure if we should do insample predictions here. What about using also the `test_size` param here? We could do also first a train-test-split like in the method with feature-importances. Then we fit the forecasters on the train data and we check the inverse variance for the predictions only?

Reply:
also I think some forecasters can not completely predict all insample values, they might raise an error.

Reply:
> I am not sure if we should do insample predictions here. What about using also the test_size param here? We could do also first a train-test-split like in the method with feature-importances. Then we fit the forecasters on the train data and we check the inverse variance for the predictions only?

I see your points, we can do that.
What I was considering is that using error variance in test sample may be not stable because of limited observations. Also I didn't see many papers using this strategy and I'm not sure how it works in real-world applications. Besides, the method I implemented is usually taken as a benchmark in the literature and what I thought originally was to implement as in the literature.  And another defect of using test set you need to fit the forecasters twice, which will double computation cost.
The advantage of using your suggested method is that it is more compatible with current class design.
Anyway, you decide it.



> also I think some forecasters can not completely predict all insample values, they might raise an error.

That's true. But I don't think it's a problem that this forecaster need to consider. I think the forecaster used should produce `NA` values instead of raising error when can't give in-sample fit, and `NA` will not affect the implementation now.

Reply:
what is the variance between NaN and a value? I am fine with in-sample approach, but then we might also change the one I implemented and get rid of the train-test-split. We can do that in a separate PR and we would have to deprecate the `test_size` parameter

Reply:
> what is the variance between NaN and a value? 
`NA` will be omitted when compute statistics such as mean, variance, just like missing values.
Let's stick with the behavior now, I will use the testset to compute variance.

Reply:
yes, I think that is the cleaner and more stable way to calculate the variance :)